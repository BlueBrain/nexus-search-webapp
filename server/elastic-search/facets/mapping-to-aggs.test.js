import mappingToAggs from "./mappings-to-aggs";

describe("mappingToAggs()", () => {
  it("makes a properly formed agg object", () => {
    const input = {
      sex: {
        type: "nested",
        properties: {
          "@id": {
            type: "text",
            fields: {
              raw: {
                type: "keyword"
              }
            }
          },
          label: {
            type: "text",
            fields: {
              raw: {
                type: "keyword"
              }
            }
          }
        }
      }
    };
    expect(mappingToAggs(input)).toMatchSnapshot();
  });

  it("makes a properly formed agg object with multiple inputs", () => {
    const input = {
      sex: {
        type: "nested",
        properties: {
          "@id": {
            type: "text",
            fields: {
              raw: {
                type: "keyword"
              }
            }
          },
          label: {
            type: "text",
            fields: {
              raw: {
                type: "keyword"
              }
            }
          }
        }
      },
      species: {
        type: "nested",
        properties: {
          "@id": {
            type: "text",
            fields: {
              raw: {
                type: "keyword"
              }
            }
          },
          label: {
            type: "text",
            fields: {
              raw: {
                type: "keyword"
              }
            }
          }
        }
      }
    };
    expect(mappingToAggs(input)).toMatchSnapshot();
  });

  it("can transform even nested property types", () => {
    const input = {
      brainLocation: {
        type: "nested",
        properties: {
          brainRegion: {
            type: "nested",
            properties: {
              "@id": {
                type: "text",
                fields: {
                  raw: {
                    type: "keyword"
                  }
                }
              },
              label: {
                type: "text",
                fields: {
                  raw: {
                    type: "keyword"
                  }
                }
              }
            }
          }
        }
      }
    };
    expect(mappingToAggs(input)).toMatchSnapshot();
  });

  it("can handle distribution, which is a hardcoded special case", () => {
    const input = {
      distribution: {
        type: "nested",
        properties: {
          "@context": {
            type: "text",
            fields: {
              keyword: {
                type: "keyword",
                ignore_above: 256
              }
            }
          },
          contentSize: {
            properties: {
              unit: {
                type: "text",
                fields: {
                  keyword: {
                    type: "keyword",
                    ignore_above: 256
                  }
                }
              },
              unitCode: {
                type: "text",
                fields: {
                  keyword: {
                    type: "keyword",
                    ignore_above: 256
                  }
                }
              },
              value: {
                type: "long"
              }
            }
          },
          digest: {
            properties: {
              alg: {
                type: "text",
                fields: {
                  keyword: {
                    type: "keyword",
                    ignore_above: 256
                  }
                }
              },
              algorithm: {
                type: "text",
                fields: {
                  keyword: {
                    type: "keyword",
                    ignore_above: 256
                  }
                }
              },
              value: {
                type: "text",
                fields: {
                  keyword: {
                    type: "keyword",
                    ignore_above: 256
                  }
                }
              }
            }
          },
          downloadURL: {
            type: "text"
          },
          mediaType: {
            type: "text",
            fields: {
              raw: {
                type: "keyword"
              }
            }
          },
          originalFileName: {
            type: "text"
          }
        }
      }
    };

    expect(mappingToAggs(input)).toMatchSnapshot();
  });
});
