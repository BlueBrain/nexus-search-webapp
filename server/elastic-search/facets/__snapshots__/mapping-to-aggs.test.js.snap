// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mappingToAggs() can handle distribution, which is a hardcoded special case 1`] = `
Object {
  "distribution": Object {
    "aggs": Object {
      "mediaType": Object {
        "terms": Object {
          "field": "distribution.mediaType.raw",
        },
      },
    },
    "nested": Object {
      "path": "distribution",
    },
  },
}
`;

exports[`mappingToAggs() can transform even nested property types 1`] = `
Object {
  "brainLocation": Object {
    "aggs": Object {
      "brainRegion": Object {
        "aggs": Object {
          "@id": Object {
            "terms": Object {
              "field": "brainLocation.brainRegion.@id.raw",
            },
          },
          "label": Object {
            "terms": Object {
              "field": "brainLocation.brainRegion.label.raw",
            },
          },
        },
        "nested": Object {
          "path": "brainLocation.brainRegion",
        },
      },
    },
    "nested": Object {
      "path": "brainLocation",
    },
  },
}
`;

exports[`mappingToAggs() makes a properly formed agg object 1`] = `
Object {
  "sex": Object {
    "aggs": Object {
      "@id": Object {
        "terms": Object {
          "field": "sex.@id.raw",
        },
      },
      "label": Object {
        "terms": Object {
          "field": "sex.label.raw",
        },
      },
    },
    "nested": Object {
      "path": "sex",
    },
  },
}
`;

exports[`mappingToAggs() makes a properly formed agg object with multiple inputs 1`] = `
Object {
  "sex": Object {
    "aggs": Object {
      "@id": Object {
        "terms": Object {
          "field": "sex.@id.raw",
        },
      },
      "label": Object {
        "terms": Object {
          "field": "sex.label.raw",
        },
      },
    },
    "nested": Object {
      "path": "sex",
    },
  },
  "species": Object {
    "aggs": Object {
      "@id": Object {
        "terms": Object {
          "field": "species.@id.raw",
        },
      },
      "label": Object {
        "terms": Object {
          "field": "species.label.raw",
        },
      },
    },
    "nested": Object {
      "path": "species",
    },
  },
}
`;
